src/pages/CommentVisit.tsx

```diff
@@ -0,0 +1,128 @@
+      if (newArrayUser[0].visits.length === 0) {
+        return;
+      } else {
+        setVisitArray([newArrayUser[0].visits[0].doctor, newArrayUser[0].visits[0].id.toString()]);
+      }
+    }
+  };
+
+  const handleChangeSelectedVisitForUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newArrayVisit = e.target.value.split(' ');
+    if (newArrayVisit.length === 0 || e.target.value.length === 0) return;
+    setVisitArray(newArrayVisit);
+  };
+
+  const handleTextAreaChange = (e: React.ChangeEventHandler<HTMLTextAreaElement>) => {
+    // jak to otypowac?
+    setTextForComment(e.currentTarget.value);
+  };
+  const handleSubmitComment = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+
+    if (textForComment === '' || visitArray === undefined || selectedVisitForUser === undefined)
+      return;
+
+    usersVisits.map((el) => {
+      if (el.name === selectedUser) {
+        return el.visits.map((el) => {
+          if (el.id === Number(visitArray[1]) && el.doctor === visitArray[0]) {
+            el.comment = textForComment;
+            //ten zapis nic nie zmienial, jak to lepiej zapisac?
+            return { ...el, comment: textForComment };
+          }
+        });
+      }
+    });
+
+    setSelectedUser('');
+    setSelectedVisitForUser([]);
+    setVisitArray([]);
+    setTextForComment('');
+  };
+  return (
+    <>
+      <Box sx={mainBoxSxStyle}>
+        <form>
+          <SingleSelect
+            text={'Select user:'}
+            textPlaceholder={'list of users'}
+            arrayProps={databaseUsersVisits}
+            //?? jak mu pomoc w ts?
+            onChangeProp={handleChangeSelectedUser}
+            valueProp={selectedUser}
+          />
+
+          <Flex sx={FlexForLabelSxStyle}>
+            <Box sx={labelSxStyle}>
+              <FormLabel>Select visit: </FormLabel>
+            </Box>
+            <Box>
+              <Select
+                onChange={handleChangeSelectedVisitForUser}
+                placeholder='list of visit'
+                // zaznaczanie nie dziala na stronie, nie pokazuje wybranego elementu z listy
+                value={visitArray}
+              >
+                {selectedVisitForUser !== undefined &&
+                  selectedVisitForUser.map((el) => {
+                    if (el.date < new Date() && el.comment.length === 0)
+                      return (
+                        <option key={el.id} value={el.doctor + ' ' + el.id}>
+                          visit: #{el.id} doctor: {el.doctor}
+                        </option>
+                      );
+                  })}
+              </Select>
+            </Box>
+          </Flex>
+          {/* chcialem zastapic to powyzej, ponizszym zapisem ale mi wywala blad */}
+          {/* react-dom.development.js:86 Warning: The `value` prop supplied to <select> must be a scalar value if `multiple` is false. */}
+          {/* <SingleSelectForCombinedArrays
```

Comment:
value do select w chakrze muszą być w odpowiedniej formie z tego co pamiętam, może przez to

Reply:
mysle ze on placze o te warunki:
      arrayProps={selectedVisitForUser !== undefined ? selectedVisitForUser : []}
            valueProp={visitArray !== undefined ? visitArray : ['']}